src/NotificationProcessor/BaseProcessor.php

```diff
@@ -0,0 +1,98 @@
+<?php declare(strict_types=1);
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2021 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <shopware@adyen.com>
+ */
+
+namespace Adyen\Shopware\NotificationProcessor;
+
+use Adyen\Shopware\Entity\Notification\NotificationEntity;
+use Shopware\Core\Checkout\Order\Aggregate\OrderTransaction\OrderTransactionStateHandler;
+use Shopware\Core\Checkout\Order\OrderEntity;
+use Shopware\Core\Checkout\Payment\PaymentMethodEntity;
+
+class BaseProcessor implements NotificationProcessorInterface
+{
+    /**
+     * @var OrderEntity
+     */
+    protected $order;
+    /**
+     * @var NotificationEntity
+     */
+    protected $notification;
+    /**
+     * @var OrderTransactionStateHandler
+     */
+    protected $transactionStateHandler;
+
+    public function process()
```

Comment:
is it needed? Should this may be in an interface instead?

Reply:
it's in the interface, that's why I have to implement it here.

Reply:
I think you can move the interface implementation to the final classes and the abstract class doesn't need to have it implemented

Reply:
Ah I see your idea, I should convert this class to an abstract class and create another concrete class that will serve as the generic notification processor (for all the other event codes) and then only the concrete classes will implement the interface and extend this abstract class.